### PR TITLE
Update REST API yamls

### PR DESCRIPTION
**Pull request by dulanjalidilmi**
## Purpose

This PR updates the following REST API yamls:
- admin-v2.yaml
- publisher-v2.yaml
- devportal-v2.yaml 
- gateway-v1.yaml

## Todo
https://github.com/wso2/docs-apim/issues/5972

### CODE REVIEW

File: en/docs/reference/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml [hunk 1]
```diff
@@ -284,10 +313,187 @@ paths:
...
+
+    #-----------------------------------------------------
+    # Get the subscription information of an api by providing the api context and version.
+    #-----------------------------------------------------
+    get:
+      x-code-samples:
+        - lang: Curl
+          source: 'curl -k -X GET "https://127.0.0.1:9443/api/am/gateway/v2/apis/7a2298c4-c905-403f-8fac-38c73301631f?tenantDomain=registry.com"
+             -H "Authorization: Basic YWRtaW46YWRtaW4="'
+      summary: Get the subscription information of an api by providing the api uuid.
```
Comment (ruthryi) [inline en/docs/reference/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml:377]: Change this to "API"

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml [hunk 2]
```diff
@@ -1490,7 +1525,7 @@ paths:
         - API Revisions
       summary: Retrieve Revision
       description: |
-        Retrieve a revision of an API
+        Retrieve a revision of an API (This resource is not supported at the moment)
```
Comment (ruthryi) [inline en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:1528]: Suggestions : 
1. This is resource is not supported in the current release.
2. This resource is not supported by WSO2 API Manager as of yet.

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml [hunk 3]
```diff
@@ -3336,104 +3421,129 @@ paths:
...
           description: |
             OK.
-            mediation policy uploaded
+            Successful response with updated WSDL definition
           headers:
             ETag:
               description: |
-                Entity Tag of the response resource.
-                Used by caches, or in conditional requests (Will be supported in future).
+                Entity Tag of the response resource. Used by caches, or in conditional requests (Will be supported in future).
```
Comment (ruthryi) [inline en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:3546]: Please see previous suggestions mentioning future support.

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml [hunk 4]
```diff
@@ -3444,226 +3554,173 @@ paths:
...
+          -F file=@api.wsdl "https://127.0.0.1:9443/api/am/publisher/v3/apis/96077508-fd01-4fae-bc64-5de0e2baf43c/wsdl"'
 
-  /apis/{apiId}/mediation-policies/{mediationPolicyId}:
+  /apis/{apiId}/graphql-policies/complexity:
     get:
       tags:
-        - API Mediation Policy
-      summary: Get an API Specific Mediation Policy
+        - GraphQL Policies
+      summary: Get the Complexity Related Details of an API
```
Comment (ruthryi) [inline en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:3587]: Should be "complexity-related details"

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml [hunk 5]
```diff
@@ -3444,226 +3554,173 @@ paths:
...
       description: |
-        This operation can be used to retrieve a particular API specific mediation policy.
-      operationId: getAPIMediationPolicyByPolicyId
+        This operation can be used to retrieve complexity related details belonging to an API by providing the API id.
```
Comment (ruthryi) [inline en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:3589]: Should be "complexity-related details"
Also, "API ID". Kindly check for "ID" in other descriptions as well.

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml [hunk 6]
```diff
@@ -3444,226 +3554,173 @@ paths:
...
+      operationId: getGraphQLPolicyComplexityOfAPI
 
-    delete:
+
+    put:
       tags:
-        - API Mediation Policy
-      summary: Delete an API Specific Mediation Policy
+        - GraphQL Policies
+      summary: Update Complexity Related Details of an API
```
Comment (ruthryi) [inline en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:3626]: Should be "complexity-related details"

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml [hunk 7]
```diff
@@ -3444,226 +3554,173 @@ paths:
...
       description: |
-        This operation can be used to delete an existing API specific mediation policy providing the Id of the API and the Id of the mediation policy.
-      operationId: deleteAPIMediationPolicyByPolicyId
+        This operation can be used to update complexity details belonging to an API by providing the id of the API.
```
Comment (ruthryi) [inline en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:3628]: - "by providing the API ID." sounds better

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml [hunk 8]
```diff
@@ -3672,308 +3729,387 @@ paths:
...
               description: |
-                Entity Tag of the response resource. Used by caches, or in conditional requests (Will be supported in future).
+                Entity Tag of the blocked subscription.
+                Used by caches, or in conditional requests (Will be supported in future).
               schema:
                 type: string
             Last-Modified:
               description: |
-                Date and time the resource has been modifed the last time.
+                Date and time the subscription has been blocked.
```
Comment (ruthryi) [inline en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:3865]: "was blocked" gives the correct meaning IMO. Please confirm.

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml [hunk 9]
```diff
@@ -3672,308 +3729,387 @@ paths:
...
                 Used by caches, or in conditional requests (Will be supported in future).
```
Comment (ruthryi) [inline en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:3866]: See previous comment of suggestions for future support

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml [hunk 10]
```diff
@@ -3672,308 +3729,387 @@ paths:
...
+          source: 'curl -k -X POST -H "Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8"
+          "https://127.0.0.1:9443/api/am/publisher/v3/apis/96077508-fd01-4fae-bc64-5de0e2baf43c/publish-to-external-stores?externalStoreId=Store123#"'
+
+  /apis/{apiId}/generate-key:
+    post:
+      tags:
+        - APIs
+      summary: Generate internal API Key to invoke APIS.
+      description: |
+        This operation can be used to generate internal api key which used to invoke API.
```
Comment (ruthryi) [inline en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:3892]: "API"

Kindly check in the other descriptions as well.

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml [hunk 11]
```diff
@@ -3672,308 +3729,387 @@ paths:
...
+            Format of output documents. Can be YAML or JSON.
+          schema:
+            type: string
+            enum:
+              - JSON
+              - YAML
+        - name: preserveStatus
+          in: query
+          description: |
+            Preserve API Status on export
```
Comment (ruthryi) [inline en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:3967]: "Preserve API Status during export"

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml (hunk 8 above)
Comment (dulanjalidilmi) [reply en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:3865]: How about "Date and time of the subscription which was blocked."

File: en/docs/reference/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml (hunk 1 above)
Comment (dulanjalidilmi) [reply en/docs/reference/product-apis/gateway-apis/gateway-v1/gateway-v1.yaml:377]: Fixed

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml (hunk 2 above)
Comment (dulanjalidilmi) [reply en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:1528]: Fixed

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml (hunk 3 above)
Comment (dulanjalidilmi) [reply en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:3546]: Fixed.

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml (hunk 4 above)
Comment (dulanjalidilmi) [reply en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:3587]: Fixed.

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml (hunk 5 above)
Comment (dulanjalidilmi) [reply en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:3589]: Fixed.

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml (hunk 6 above)
Comment (dulanjalidilmi) [reply en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:3626]: Fixed

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml (hunk 7 above)
Comment (dulanjalidilmi) [reply en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:3628]: Fixed.

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml (hunk 9 above)
Comment (dulanjalidilmi) [reply en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:3866]: Fixed.

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml (hunk 10 above)
Comment (dulanjalidilmi) [reply en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:3892]: Fixed

File: en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml (hunk 11 above)
Comment (dulanjalidilmi) [reply en/docs/reference/product-apis/publisher-apis/publisher-v2/publisher-v2.yaml:3967]: Fixed

**Review by ruthryi — APPROVED** [review]

**Pull request merged**